### PR TITLE
periodic dihedral extrapolation

### PR DESCRIPTION
**Pull request by marvinbernhardt**
When dealing with dihedrals I ran into problems with the extrapolation. potential_to_gromacs.sh would extrapolate both sides to 180°. If the new potential is then interpolated and the force is calculated, one obtains a very high force for x=180. 

This pull request is the solution I came up with. It will ignore the gradient of the last points on the right side and extrapolate linearly to the same value one finds on the left side.

### CODE REVIEW

**Comment by junghans** [comment]
So we set the force to same value on both side (https://github.com/votca/csg/blob/master/share/scripts/inverse/table_to_xvg.pl#L101)
but the value is too high, correct?

File: share/scripts/inverse/potential_extrapolate.sh [hunk 1]
```diff
@@ -90,9 +90,12 @@ intermediate="$(critical mktemp "${input}.onlyleft.XXXXX")"
 if [[ $pot_type = "non-bonded"  ]]; then
   do_external table extrapolate --function ${lfct:-exponential} --avgpoints $avg_points --region left "${input}" "${intermediate}"
   do_external table extrapolate --function ${rfct:-constant} --avgpoints 1 --region right "${intermediate}" "${output}"
-elif [[ $pot_type = "bond"  || $pot_type = "angle" || $pot_type = "dihedral" ]]; then
+elif [[ $pot_type = "bond"  || $pot_type = "angle" ]]; then
   do_external table extrapolate --function ${lfct:-linear} --avgpoints $avg_points --region left "${input}" "${intermediate}"
   do_external table extrapolate --function ${rfct:-linear} --avgpoints $avg_points --region right "${intermediate}" "${output}"
+elif [[ $pot_type = "dihedral" ]]; then
+  do_external table extrapolate --function ${lfct:-linear} --avgpoints $avg_points --region left "${input}" "${intermediate}"
```
Comment (junghans) [inline share/scripts/inverse/potential_extrapolate.sh]: Why would you not use `periodic` on the left side as well?

File: share/scripts/inverse/potential_extrapolate.sh [hunk 2]
```diff
@@ -90,9 +90,12 @@ intermediate="$(critical mktemp "${input}.onlyleft.XXXXX")"
...
+  do_external table extrapolate --function ${rfct:-periodic} --avgpoints $avg_points --region right "${intermediate}" "${output}"
```
Comment (junghans) [inline share/scripts/inverse/potential_extrapolate.sh]: You will need to fix the help above in this script as well.

**Comment by junghans** [comment]
@marvinbernhardt I am all for merging this as dihedrals aren't very well supported currently. Have a look at my comments above. Do dihedrals always need to go from 0 to 180?

**Comment by junghans** [comment]
@votca-bot changelog: add periodic extrapolation for dihedrals

**Comment by marvinbernhardt** [comment]
> So we set the force to same value on both side (https://github.com/votca/csg/blob/master/share/scripts/inverse/table_to_xvg.pl#L101)
> but the value is too high, correct?

Yes, precisely

File: share/scripts/inverse/potential_extrapolate.sh (hunk 1 above)
Comment (marvinbernhardt) [reply share/scripts/inverse/potential_extrapolate.sh]: So the scheme I have here is to extrapolate the left side normally and then extrapolate the right side to meet leftmost value. I think this scheme should suffice in most cases. A more elaborate scheme would be to pre-extrapolate on both sides, save the value at the boundaries, and extrapolate both sides to their average. But that would require a restructuring of the whole script and possibly others.

**Comment by marvinbernhardt** [comment]
Ups, sorry reverted those codacy changes. Will change again.

**Comment by junghans** [comment]
@marvinbernhardt I undid the last two commits, was there a special reason to split the option parsing? I think we can alllow periodic for the left as well, even though we don't have a use case yet.

**Review by junghans — APPROVED** [review]

**Pull request merged**